modules/planning/tasks/deciders/open_space_fallback_decider.h

```diff
@@ -20,18 +20,43 @@
 
 #pragma once
 
+#include <algorithm>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "Eigen/Dense"
+#include "modules/common/configs/proto/vehicle_config.pb.h"
+#include "modules/common/configs/vehicle_config_helper.h"
+#include "modules/common/math/vec2d.h"
+#include "modules/planning/common/frame.h"
+#include "modules/planning/common/obstacle.h"
+#include "modules/planning/common/planning_gflags.h"
+#include "modules/planning/common/trajectory/discretized_trajectory.h"
+#include "modules/planning/common/trajectory/publishable_trajectory.h"
 #include "modules/planning/tasks/deciders/decider.h"
 
 namespace apollo {
 namespace planning {
-
 class OpenSpaceFallbackDecider : public Decider {
  public:
   explicit OpenSpaceFallbackDecider(const TaskConfig& config);
 
  private:
   apollo::common::Status Process(
       Frame* frame) override;
+
+  // bool IsCollisionFreeTrajectory(const ADCTrajectory& trajectory_pb);
+
+  void BuildPredictedEnvironment(const std::vector<const Obstacle*>& obstacles);
+
+  bool IsCollisionFreeTrajectory(const TrajGearPair& trajectory_pb);
+
+ private:
+  std::vector<std::vector<common::math::Box2d>> predicted_bounding_rectangles_;
+
+  double collision_distance_;
```

Comment:
I think It's better pass the collision_distance_  and predicted_bounding_rectangles_ between "check collision"  and "process" rather than put in private data field

Reply:
> I think It's better pass the collision_distance_ and predicted_bounding_rectangles_ between "check collision" and "process" rather than put in private data field
your four comments are fixed